### PR TITLE
Refactor replace errors wrap with join

### PR DESCRIPTION
**Pull request by ykadowak**
### Description:
Replaced `errors.Wrap` with `errors.Join`. Made some modifications to several tests accordingly.

### Versions:

- Go Version: 1.20.3
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 2.0.9

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

### Special notes for your reviewer:

This will be done in another PR.
https://github.com/vdaas/vald/blob/e2a72fe2c71a9195656041c3c6d9205c7497c0c4/internal/net/grpc/stream.go#L75-L78

### CODE REVIEW

File: internal/net/grpc/stream.go [hunk 1]
```diff
@@ -72,6 +72,9 @@ func BidirectionalStream[Q any, R any](ctx context.Context, stream ServerStream,
 			if errs == nil {
 				errs = err
 			} else {
+				// FIXME: It should have been errors.Join here, but it remains errors.Wrap because the bench/agent/stream tests will fail.
+				// The reason is likely that when it's changed to errors.Join, status.ParseError cannot parse errors correctly,
+				// but this needs to be investigated further.
```
Comment (vankichi) [inline internal/net/grpc/stream.go:77]: :+1:

**Review by vankichi — APPROVED** [review]
LGTM! Thanks :smile: 
Welcome to Vald world :tada:

**Review by kpango — APPROVED** [review]
LGTM thanks

**Comment by kpango** [comment]
@all-contributors please add @ykadowak for code,test

**Pull request merged**